src/kixi/comms/components/kinesis.clj

```diff
@@ -182,7 +185,11 @@
 
   (send-command! [{:keys [producer-in-ch]} command version user payload opts]
     (when producer-in-ch
-      (async/put! producer-in-ch [:command command version user payload opts])))
+      (async/put! producer-in-ch [:command command version user payload 
+                                  (assoc opts
+                                          :command-id
+                                          (or (:kixi.comms.command/id payload)
```

Comment:
The 'payload' at this point isn't just the message, it has our wrapping stuff around it, where :kixi.comms.command/id will be for commands and events.